en/docs/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/general-recommendations-for-secure-coding.md

```diff
@@ -574,6 +801,204 @@ HTML Purifier[^19] should be used to sanitize HTML content. Default HTML Purifie
 !!! tip hint important "WSO2 Document Reference"
     Further information on required changes and recommended configuration for WSO2 products as well as production deployments are available at [Engineering Guidelines - Security Related HTTP Headers](../security-related-http-headers.md).
 
+### Go Specific Recommendations
+
+#### Output Encoding
+`html/template` package can be used for HTML templating as it provides automatic contextual encoding. For manual encoding, use `html.EscapeString` function to convert special characters to HTML entities.
+
+!!! bug error "Example Incorrect Usage"
+    ```go
+    import "text/template"
+    
+    tmpl := `<div>{{.UserInput}}</div>`
+    t := template.Must(template.New("example").Parse(tmpl))
+    t.Execute(w, data)
+    ```
+
+!!! success check done "Example Correct Usage"
+    ```go
+    import "html/template"
+    
+    tmpl := `<div>{{.UserInput}}</div>`
+    t := template.Must(template.New("example").Parse(tmpl))
+    t.Execute(w, data)
+    ```
+
+For manual encoding in non-template contexts
+
+!!! bug error "Example Incorrect Usage"
+    ```go
+    fmt.Fprintf(w, `<a href="%s">Link</a>`, userInput)
+    ```
+
+!!! success check done "Example Correct Usage"
+    ```go
+    import "html"
+    
+    fmt.Fprintf(w, `<a href="%s">Link</a>`, html.EscapeString(userInput))
+    ```
+
+Encode for JavaScript contexts
+
+!!! bug error "Example Incorrect Usage"
+    ```go
+    fmt.Fprintf(w, `<script>var data = "%s";</script>`, userInput)
+    ```
+
+!!! success check done "Example Correct Usage"
+    ```go
+    import (
+        "encoding/json"
+        "html/template"
+    )
+    
+    jsData, err := json.Marshal(userInput)
+    if err != nil {
+        http.Error(w, "Encoding error", http.StatusInternalServerError)
+        return
+    }
+
+    tmpl := `<script>var data = {{.JSData}};</script>`
+    t := template.Must(template.New("js").Parse(tmpl))
+    t.Execute(w, map[string]template.JS{"JSData": template.JS(jsData)})
+    ```
+
+Encode URLs
+
+!!! bug error "Example Incorrect Usage"
+    ```go
+    fmt.Fprintf(w, `<a href="%s">Click Here</a>`, userInput)
+    ```
+
+!!! success check done "Example Correct Usage"
+    ```go
+    escapedURL := template.URLQueryEscaper(userInput)
+    fmt.Fprintf(w, `<a href="%s">Click Here</a>`, escapedURL)
+    ```
+
+#### Output Sanitization
+
+For Go applications, established libraries such as `bluemonday` can be used to sanitize HTML content. The `bluemonday` library allows you to define a policy for allowed HTML elements and attributes, which helps in preventing XSS attacks.
```

Comment:
Let's discuss if there is actually a need for HTML sanitization in new code bases anymore. Ideally the recommendation here should be that we should use safer options like markdown if there is any such requirement, and not to use HTML inputs (since we are talking about a new line of products, we can use safer options)